app/models/spree/order.rb

```diff
@@ -171,6 +171,11 @@ def amount
       line_items.inject(0.0) { |sum, li| sum + li.amount }
     end
 
+    # Order total without any applied discounts from vouchers
+    def pre_discount_total
```

Comment:
In case we have other types of "discount" in the system, I think a more specific name should be used here:

```suggestion
```

Reply:
Hmmm interesting! I don't really associate the word "discount" with the idea of percentages specifically, just with a reduction in price.

Yeah I guess at the moment the concepts "discount" and "voucher" are kind of synonymous in the codebase, vouchers apply a discount and the only kind of discount is from vouchers. But we have the `Voucher` class, and voucher objects that are saved on a _distributor_ and define a kind of "voucher rule", then we have voucher adjustments which are a different object altogether and are saved on an order.

As always, [naming things is hard](https://hilton.org.uk/blog/why-naming-things-is-hard).

I guess we could say that "voucher" is the _form_ and "discount" is the _function_? I think what the method is trying to represent is the **price** of the order prior to discounts that might modify that price, so conceptually I'm leaning more towards the function than the form in this case?

Hmmm so I'm not sure I completely agree, but then again I also don't completely disagree! :sweat_smile: 

Maybe the other devs have some opinions one way or the other on this naming conundrum..?

Reply:
When I reviewed the pull request I also stumbled over `pre_discount_total` because the name seemed very general while we're only looking at vouchers at the moment. Being more specific can help people understand better what's happening. So, I would have named it like @dacook.

But I didn't raise this because a more general name can be more resilient to change later. I don't have the association of percentages and discount. I find the general concept of "pre discount" quite fitting.

So while I would have named it differently in the first place, I didn't think it was worth changing now. Happy either way but I personally wouldn't invest more work in this. :shrug: :cat2: :zzz:

Reply:
I would keep as it is, even though "vouchers" are the only possible discount right now it still convey an understandable meaning .

Reply:
Thanks for the discussion! Yes on further thought, "discounts" aren't necessarily a percentage. In my mind now, a "discount" is a price reduction provided by the store (whether flat or percentage). Whereas I imagined that a voucher could also be a gift from another person (eg I pay $10 for a gift voucher, for someone else to spend at the store). 

So I agree about not changing this now, we can discuss if/when we need to change it!